### PR TITLE
Add shardingsphere-scenario

### PR DESCRIPTION
**Pull request by tuohai666**
Please fill up those following information before send Pull request 

* Pull request NO: #1934 (https://github.com/apache/incubator-skywalking/pull/1934)
* Agent repository URL: https://github.com/tuohai666/incubator-skywalking
* Testcases repository URL: https://github.com/tuohai666/skywalking-agent-testcases

### CODE REVIEW

**Comment by ascrutae** [comment]
There are three faults in this test case:

1. The test case container miss the MySQL environment variable 
2. The MySQL environment variables is not read by test case. It use the default value and those value cannot run in container environment, it will happen exception that the network cannot reach.
2. There are lots of fields value is incorrect. For example: spanId , parentSpanID

File: shardingsphere-scenario/config/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - mock-collector
+      - mysql-server
+    volumes:
+     - {AGENT_FILE_PATH}:/usr/local/shardingsphere-scenario/agent
+    depends_on:
+      - mock-collector
+      - mysql-server
+
+  mock-collector:
+    image: skywalking/skywalking-mock-collector:{COLLECTOR_IMAGE_VERSION}
```
Comment (ascrutae) [inline shardingsphere-scenario/config/docker-compose.yml]:
```suggestion
    image: skyapm/mock-collector:{COLLECTOR_IMAGE_VERSION}
```

File: shardingsphere-scenario/config/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+version: '2.1'
+services:
+  mysql-server:
```
Comment (ascrutae) [inline shardingsphere-scenario/config/docker-compose.yml]: I suggest that you should remove `mysql-server` configure item if you don't use mysql

**Review by ascrutae — COMMENTED** [review]
I don't see any segment validate message about `/shardingsphere-case/case/execute` entry service of  `ShardingSphere` scenario in expected.yaml file, Is there you miss that? I suggest that you should add those validate message.

File: shardingsphere-scenario/config/expectedData.yaml [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            peer: ''
+            peerId: '0'
+          - operationName: "/case/execute"
+            operationId: '0'
+            parentSpanId: "-1"
+            spanId: '0'
+            spanLayer: Http
+            tags:
+              - key: url
+                value: http://localhost:18080/shardingsphere-case/case/execute
```
Comment (ascrutae) [inline shardingsphere-scenario/config/expectedData.yaml]:
```suggestion
                value: http://localhost:{SERVER_OUTPUT_PORT}/shardingsphere-case/case/execute
```

File: shardingsphere-scenario/config/expectedData.yaml [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            spanId: "0"
+            spanLayer: Unknown
+            refs:
+              - parentEndpointId: 0
+                parentEndpoint: "/ShardingSphere/JDBCRootInvoke/"
+                networkAddressId: 0
+                entryEndpointId: 0
+                refType: CrossThread
+                parentSpanId: 0
+                parentTraceSegmentId: not null
```
Comment (ascrutae) [inline shardingsphere-scenario/config/expectedData.yaml]: I suggest that you should make sense the value of `parentTraceSegmentId`

File: shardingsphere-scenario/config/expectedData.yaml [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
```
Comment (ascrutae) [inline shardingsphere-scenario/config/expectedData.yaml]: I suggest that you should make sense the value of `parentTraceSegmentId`

**Review by ascrutae — COMMENTED** [review]
@tuohai666 I can't see any information about data sharding or database orchestration from expected data. Is that miss this information in this scenario?

**Comment by tuohai666** [comment]
@ascrutae Thanks for your comments. The test case is a single routing scenario, and the expected data is same as actual data.
> @tuohai666 I can't see any information about data sharding or database orchestration from expected data. Is that miss this information in this scenario?

**Comment by wu-sheng** [comment]
> Thanks for your comments. The test case is a single routing scenario, and the expected data is same as actual data.

@tuohai666 Is single DB routing scenario enough? This case is designed to verify the plugin and benefit for shardingsphere ecosystem. We can't make that decision, but I think as a routing lib, it should test at least for two DBs.
@terrymanu @tristaZero What do you think? Or prefer?

**Comment by tuohai666** [comment]
@wu-sheng Thanks for quick reply. I have tried  full routing before this version of test case. The problem is that full routing can not guarantee the order of the executing threads, and leads to an unpredictable spans order.

File: shardingsphere-scenario/config/expectedData.yaml (hunk 4 above)
Comment (tuohai666) [reply shardingsphere-scenario/config/expectedData.yaml]: The parent trace segmentId is unpredictable. During the last test is segmentId is .1.15562751912310010. Any suggestion?

File: shardingsphere-scenario/config/expectedData.yaml (hunk 3 above)
Comment (tuohai666) [reply shardingsphere-scenario/config/expectedData.yaml]: fixed

**Comment by terrymanu** [comment]
If SkyWalking's test framework can assert for concurrency(ignore execute's spans order), I prefer use two DBs scenario.
If not, use single DB is fine too. From technical layer, single DB and two DBs are really similar (beside a new thread and merge engine, but merge is not monitored by SkyWalking)

File: shardingsphere-scenario/config/expectedData.yaml [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+            spanId: '0'
+            spanLayer: Unknown
+            refs:
+              - parentEndpointId: 0
+                parentEndpoint: "/case/execute"
+                networkAddressId: 0
+                entryEndpointId: 0
+                refType: CrossThread
+                parentSpanId: 2
+                parentTraceSegmentId: not null
```
Comment (ascrutae) [inline shardingsphere-scenario/config/expectedData.yaml]:
```suggestion
                parentTraceSegmentId: "${Your_ApplicationName[3]}"
```

File: shardingsphere-scenario/config/expectedData.yaml [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
```
Comment (ascrutae) [inline shardingsphere-scenario/config/expectedData.yaml]:
```suggestion
                parentTraceSegmentId: "${Your_ApplicationName[3]}"
```

File: shardingsphere-scenario/config/expectedData.yaml [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
```
Comment (ascrutae) [inline shardingsphere-scenario/config/expectedData.yaml]:
```suggestion
                parentTraceSegmentId: "${Your_ApplicationName[3]}"
```

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**